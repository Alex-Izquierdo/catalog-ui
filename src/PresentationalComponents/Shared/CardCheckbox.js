import React from 'react';
import propTypes from 'prop-types';
import { Checkbox } from '@patternfly/react-core';

const CardCheckbox = ({ handleCheck, isChecked, id }) => (
  <div style={ { float: 'right' } }>
    <Checkbox
      checked={ isChecked }
      onChange={ handleCheck }
      onClick={ event => event.stopPropagation() }
      aria-label="card checkbox"
      id={ id }
    />
  </div>
);

CardCheckbox.propTypes = {
  handleCheck: propTypes.func,
  isChecked: propTypes.bool,
  id: propTypes.string
};

export default CardCheckbox;
