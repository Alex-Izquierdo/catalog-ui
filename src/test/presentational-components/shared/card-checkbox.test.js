import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CardCheckbox from '../../../PresentationalComponents/Shared/CardCheckbox';

describe('<CardCheckbox />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      handleCheck: jest.fn(),
      isChecked: false,
      id: 'foo-id'
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<CardCheckbox { ...initialProps }/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleCheck on checkbock click', () => {
    const wrapper = mount(<CardCheckbox { ...initialProps }/>);
    wrapper.find('input').simulate('change');
    expect(initialProps.handleCheck).toHaveBeenCalledWith(false, expect.any(Object));
  });
});
