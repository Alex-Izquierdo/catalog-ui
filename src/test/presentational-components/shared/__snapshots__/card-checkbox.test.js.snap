// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCheckbox /> should render correctly 1`] = `
<Checkbox
  aria-label="card checkbox"
  className=""
  id="foo-id"
  isChecked={false}
  isDisabled={false}
  isValid={true}
  onChange={[MockFunction]}
  onClick={[Function]}
  ouiaSafe={true}
/>
`;
